### PR TITLE
luxcorerender: 2.0 -> 2.4

### PR DESCRIPTION
**Pull request by jmpunkt**
###### Motivation for this change

Fix the build of the package which currently fails on master and 20.09. Bumped to a new version since this version does not require OpenCL nor Cuda to be present during compile time. Both are detected dynamically during runtime. However, the executables must be compiled such that OpenCL and Cuda are potentially supported. If OpenCL or Cuda is not present during runtime, the CPU is used as a fallback.

Notice: I was not able to verify that OpenCL or Cuda works. I do not have a working configuration for OpenCL and do not own a Nvidia card. The software runs fine even if OpenCL and Cuda are not detected.

Tested the UI executable of LuxCore with
```sh
result/bin/luxcoreui $LUXCORE_SRC/scenes/psor-cube/psor-cube.cfg
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/luxcorerender/default.nix [hunk 1]
```diff
@@ -1,63 +1,61 @@
...
-     flex libX11 libpthreadstubs python libXdmcp libxkbcommon
-     epoxy at-spi2-core dbus doxygen
-     # needed for GSETTINGS_SCHEMAS_PATH
-     gsettings-desktop-schemas glib gtk3
-     # needed for XDG_ICON_DIRS
-     gnome3.adwaita-icon-theme
-     makeWrapper
-     (stdenv.lib.getLib dconf)
-   ] ++ stdenv.lib.optionals withOpenCL [opencl-headers ocl-icd opencl-clhpp];
+  nativeBuildInputs = [ pkgconfig cmake doxygen wrapGAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/luxcorerender/default.nix:31]:
```suggestion
  nativeBuildInputs = [ pkg-config cmake doxygen wrapGAppsHook ];
```

File: pkgs/tools/graphics/luxcorerender/default.nix [hunk 2]
```diff
@@ -1,63 +1,61 @@
...
+    boostWithPython.dev
+    python3
+    embree
+    openexr
+    openimagedenoise
+    tbb
+    c-blosc
+    gtk3
+    pcre
+
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/luxcorerender/default.nix:48]:
```suggestion
```

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101413` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>luxcorerender</li>
  </ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]
Can you squash the commits together? Otherwise I think this is fine.

**Comment by dotlambda** [comment]
@jmpunkt Can you resolve the merge conflict?

**Comment by jmpunkt** [comment]
Total of three different external changes were made since this PR was created.

Took care of
- treewide: pkg-conf
- treewide: stdenv.lib
- treewide: cmake buildInputs to nativeBuildInputs

**Comment by dotlambda** [comment]
Thanks a lot @jmpunkt!

**Pull request merged**